src/test/test-fs-util.c

```diff
@@ -614,6 +614,77 @@ static void test_fsync_directory_of_file(void) {
         assert_se(fsync_directory_of_file(fd) >= 0);
 }
 
+static void test_rename_noreplace(void) {
+
+        const char* const table[] = {
+                "/reg",
+                "/dir",
+                "/fifo",
+                "/socket",
+                "/symlink",
+                NULL
+        };
+
+        _cleanup_(rm_rf_physical_and_freep) char *z = NULL;
+        const char *e, *j;
+        char **a, **b;
+
+        e = getenv("RENAME_NOREPLACE_DIR");
+        if (e) {
+                j = strjoina(e, "/testXXXXXX");
+
+                assert_se(mkdtemp_malloc(j, &z) >= 0);
+        } else
+                assert_se(mkdtemp_malloc(NULL, &z) >= 0);
```

Comment:
Maybe intialize `j` to NULL above and combine the two mkdtemp_malloc() statements here?